docs: add documentation about timestamp data type (#6853)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/type-coercion.md

  Valid boolean values are `true`, `false`, `yes`, `no`, or any substring of
  these values that starts with the first character, for example, `fal`, `y`.
  Comparison is case-insensitive.
* A `STRING` literal containing an ISO-8601 formatted datestring can be coerced

>   to a `TIMESTAMP`. A datestring containing a timezone will be converted to UTC.

Reviewer: ```suggestion
  to a `TIMESTAMP`. A datestring containing a timezone is converted to UTC.
```
